scheduler/remote_server.py

```diff
@@ -0,0 +1,487 @@
+        else:
+            time -= datetime.timedelta(days=1)
+
+            yyyymmdd = time.strftime("%Y%m%d")
+            hhmm = time.strftime("%H:%M")
+
+            new_relevant_lines = scd_util.get_relevant_lines(yyyymmdd, hhmm)
+            if not new_relevant_lines:
+                msg = "Error in schedule: Could not find any relevant_lines. Either no lines exist "
+                "or an infinite duration line could not be found."
+
+                raise ValueError(msg)
+
+            lines_diff = [d for d in new_relevant_lines if d not in relevant_lines]
+
+            for line in reversed(lines_diff):
+                if line['duration'] == '-':
+                    relevant_lines.insert(0, line)
+                    found = True
+
+
+    # Since SCDUtils relevant lines will return the last line run if the supplied time
+    # is in the beginning of two lines, we then only schedule that time if it's duration is
+    # infinite, otherwise we filter it out since we missed it's time.
+    relevant_lines = [l for l in relevant_lines if l['time'] > now or l['duration'] == '-']
+    return relevant_lines
+
+
+def _main():
+    parser = argparse.ArgumentParser(description="Automatically schedules new SCD file entries")
+    parser.add_argument('--emails-filepath',required=True, help='A list of emails to send logs to')
+    parser.add_argument('--scd-dir', required=True, help='The scd working directory')
+
+    args = parser.parse_args()
+
+    scd_dir = args.scd_dir
+
+    emailer = email_utils.Emailer(args.emails_filepath)
+
+    i = inotify.adapters.Inotify()
+
+    options = rso.RemoteServerOptions()
+    site_id = options.site_id
+
+    scd_file = '{}.scd'.format(site_id)
+
+    i.add_watch(scd_dir + '/' + scd_file)
+    scd_util = scd_utils.SCDUtils(scd_file)
+
+    log_dir = "{}/logs".format(scd_dir)
+
+    if not os.path.exists(log_dir):
+        os.makedirs(log_dir)
+
+
+    def make_schedule():
+        now = datetime.datetime.utcnow()
+
+        log_time_str = now.strftime("%Y.%m.%d.%H.%M")
+        log_file = "{}/{}.log".format(log_dir, log_time_str)
+
+        try:
+            relevant_lines = get_relevant_lines(scd_util, now)
+        except ValueError as e:
+            error_msg = ("{logtime}: Unable to make schedule\n"
+                         "\t Exception thrown:\n"
+                         "\t\t {exception}\n")
+            error_msg = error_msg.format(logtime = now.strftime("%c"),
+                                            exception=str(e))
+            with open(log_file, 'w') as f:
+                f.write(error_msg)
+
+            subject = "Unable to make schedule at {}".format(site_id)
+
+            emailer.email_log(subject, log_file)
+        else:
+            timeline = convert_scd_to_timeline(relevant_lines)
+            plot_path, pickle_path = plot_timeline(timeline, scd_dir, now)
+            new_atq_str = timeline_to_atq(timeline, scd_dir, now)
+
+            with open(log_file, 'wb') as f:
```

Comment:
Could we have a line in the log file for the time of change? Could return backup time from the timeline_to_atq function as well. There will be overlapping times in this log file when rescheduling stuff so I think there should be a line for 'updated at' to indicate what did or didn't run

Reply:
I dont think there will be overlapping times since the logging is on a minute basis, but im adding it anyway. This gets sent in an email so it might make that easier to read.